### PR TITLE
Add nonce support

### PR DESCRIPTION
**Pull request by paolapog**
This pr is linked to [#39](https://github.com/epicweb-dev/epic-stack/issues/39)
For this pr I've used helmet and partially re-used what was done in kentcdodds.com project.
I've used also CSP Evaluator extension (Chrome's one) to check the current CSP.

### CODE REVIEW

File: server/index.ts [hunk 1]
```diff
@@ -26,32 +29,76 @@ app.disable('x-powered-by')
...
+        'font-src': ["'self'"],
+        'frame-src': [
+          "'self'",
+        ],
+        'img-src': [
+          "'self'",
+          'data:',
+          'user-images.githubusercontent.com',
+          'raw.githubusercontent.com',
+          ...(MODE === 'development' ? ['cloudflare-ipfs.com'] : []),
```
Comment (kentcdodds) [inline server/index.ts:63]: I think I'd like to move the images to be served by the app server instead so we don't have to include this in the csp.

File: server/index.ts [hunk 2]
```diff
@@ -26,32 +29,76 @@ app.disable('x-powered-by')
...
+        ],
+        'media-src': ["'self'", 'res.cloudinary.com', 'data:', 'blob:'],
```
Comment (kentcdodds) [inline server/index.ts:65]: Same for this, let's limit this as much as possible.

File: app/entry.server.tsx [hunk 3]
```diff
@@ -29,24 +35,33 @@ export default async function handleRequest(
...
+          const body = new PassThrough()
 					responseHeaders.set('Content-Type', 'text/html')
+          const dataEvtTransform = new Transform({
+            transform(chunk, _, callback) {
+              const string = chunk.toString()
+              const replaced = string.replace(/data-evt-/g, `nonce="${nonce}" `)
+              callback(null, replaced)
+            },
+          })
+          pipe(dataEvtTransform).pipe(body)
```
Comment (kentcdodds) [inline app/entry.server.tsx:58]: Let's not bother with this. This is something I'm doing in my own site that's unnecessary for anyone else.

File: app/root.tsx [hunk 4]
```diff
@@ -125,14 +127,16 @@ export default function App() {
 					<ThemeSwitch />
 				</div>
 				<div className="h-5" />
-				<ScrollRestoration />
-				<Scripts />
+				<ScrollRestoration nonce={nonce}/>
+				<Scripts nonce={nonce}/>
 				<script
+          nonce={nonce}
+          suppressHydrationWarning
```
Comment (kentcdodds) [inline app/root.tsx:134]: Is this necessary? It should not be.

**Review by kentcdodds — CHANGES_REQUESTED** [review]
Thanks for working on this!

File: app/root.tsx [hunk 5]
```diff
@@ -94,7 +96,7 @@ export default function App() {
 				<meta name="viewport" content="width=device-width,initial-scale=1" />
 				<Links />
 			</head>
-			<body className="flex h-full flex-col justify-between bg-night-700 text-white">
+			<body suppressHydrationWarning className="flex h-full flex-col justify-between bg-night-700 text-white">
```
[the remark this answers is not in the document]
Comment (paolapog) [reply app/root.tsx:99]: My fault, I was suppressing an error from a Chrome's extensions (I didn't find a clue yesterday), I'll remove it 💪🏼

File: server/index.ts [hunk 6]
```diff
@@ -26,32 +29,76 @@ app.disable('x-powered-by')
...
 )
-
 // Everything else (like favicon.ico) is cached for an hour. You may want to be
 // more aggressive with this caching.
 app.use(express.static('public', { maxAge: '1h' }))
 
-morgan.token('url', (req, res) => decodeURIComponent(req.url ?? ''))
 app.use(morgan('tiny'))
 
+morgan.token('url', (req, res) => decodeURIComponent(req.url ?? ''))
```
[the remark this answers is not in the document]
Comment (paolapog) [reply server/index.ts:41]: I made a `git rebase -i main` and I took the last changes: it's already present in the app

File: app/root.tsx (hunk 4 above)
Comment (kentcdodds) [reply app/root.tsx:134]: If you're getting that warning then something's wrong. This should all work without any warnings.

Comment (paolapog) [reply app/root.tsx:134]: I was looking on the other project (kentcdodds.com) and this attribute is present. Maybe there is the same problem in the current implementation
I'll try to make it work 💪🏼

Comment (paolapog) [reply app/root.tsx:134]: update: https://react.dev/reference/react-dom/client/hydrateRoot 
`In development mode, React warns about mismatches during hydration.`
I tried with `NODE_ENV=production` and it doesn't complain 👍🏼 so I think I can remove it without concerns

**Review by kentcdodds — APPROVED** [review]
Looks great. Thanks a lot!

**Pull request merged**